subsys/net/lib/nrf_cloud/src/nrf_cloud.c

```diff
@@ -26,11 +26,16 @@ LOG_MODULE_REGISTER(nrf_cloud, CONFIG_NRF_CLOUD_LOG_LEVEL);
 /* Maintains the state with respect to the cloud. */
 static volatile enum nfsm_state m_current_state = STATE_IDLE;
 
+/* Flag to indicate if a disconnect has been requested. */
+static atomic_t disconnect_requested;
+
 /* Handler registered by the application with the module to receive
  * asynchronous events.
  */
 static nrf_cloud_event_handler_t m_event_handler;
 
+K_MUTEX_DEFINE(m_state_mutex);
```

Comment:
No need to prefix global variables (actually we should not do that according to coding standard). Add `static`.

Reply:
ok, i will remove all `m_` from the file.

does `K_MUTEX_DEFINE` not actually make it static?
```@brief Statically define and initialize a mutex.```

Reply:
`Statically` refers here to static memory, not `static` keyword.